src/panels/config/integrations/integration-panels/thread/dialog-thread-dataset.ts

```diff
@@ -72,8 +72,11 @@ class DialogThreadDataset extends LitElement implements HassDialog {
       type: "thread/store_in_platform_keychain",
       payload: {
         mac_extended_address:
+          this._params?.network.dataset?.preferred_extended_address ||
           this._params!.network.routers![0]!.extended_address,
-        border_agent_id: this._params!.network.routers![0]!.border_agent_id,
+        border_agent_id:
```

Comment:
Is this needed? I think @bgoncal said iOS uses the extended mac address as the border_agent_id. My understanding is that the iOS app is only looking for `mac_extended_address` and `active_operational_dataset` currently.

Reply:
But android will use the border agent id most probably